cmd/jujud/agent/machine.go

```diff
@@ -1363,8 +1365,8 @@ func (a *MachineAgent) removeJujudSymlinks() (errs []error) {
 // running the tests and (2) get access to the *State used internally, so that
 // tests can be run without waiting for the 5s watcher refresh time to which we would
 // otherwise be restricted.
-var newDeployContext = func(st *apideployer.State, agentConfig agent.Config) deployer.Context {
-	return deployer.NewSimpleContext(agentConfig, st)
+var newDeployContext = func(config deployer.ContextConfig) (deployer.Context, error) {
```

Comment:
Silly question but is there a specific reason this needs to be a var and not just a package scoped func ?

Reply:
I'll double check, maybe not.

Reply:
No reason at all. Removed the function and just pass in the package func where this was used.

Reply:
I think in the past it changed the order of args, but now it doesn't, so no reason to keep it.

Reply:
Actually yes, it changes the return type.